### ISSUE TITLE
Compilation failure on saga handler method: parameter name duplication

### ISSUE THREAD

**Issue by fab60** [JasperFx/jasper#465]
Maybe it's more of a Lamar issue, but the generated method code for a saga handler method with 
parameters of type `IQueryObject<TypeA>` and `IQueryObject<TypeB>` does produce both a private field and a parameter name duplication issue:

Compilation failures!

CS0100: The parameter name 'queryObject' is a duplicate
CS0100: The parameter name 'queryObject' is a duplicate
CS0102: The type 'Foo_Commands_Command1' already contains a definition for '_queryObject'
CS0102: The type 'Foo_Commands_Command1' already contains a definition for '_queryObject'
CS0229: Ambiguity between 'Foo_Commands_Command1._queryObject' and 'Foo_Commands_Command1._queryObject'
CS0229: Ambiguity between 'IQueryObject<TypeA>' and 'IQueryObject<TypeB>'
CS0229: Ambiguity between 'Foo_Commands_Command1._queryObject' and 'Foo_Commands_Command1._queryObject'
CS0229: Ambiguity between 'IQueryObject<TypeA>' and 'IQueryObject<TypeB>'
CS0229: Ambiguity between 'Foo_Commands_Command1._queryObject' and 'Foo_Commands_Command1._queryObject'
CS0229: Ambiguity between 'IQueryObject<TypeA>' and 'IQueryObject<TypeA>'
CS0229: Ambiguity between 'Foo_Commands_Command1._queryObject' and 'Foo_Commands_Command1._queryObject'
CS0229: Ambiguity between 'Foo_Commands_Command1._queryObject' and 'Foo_Commands_Command1._queryObject'
CS0229: Ambiguity between 'Foo_Commands_Command1._queryObject' and 'Foo_Commands_Command1._queryObject'

Code:

```  // START: Foo_Commands_Command1
    public class Foo_Commands_Command1 : Jasper.Messaging.Model.MessageHandler
    {
        private readonly Jasper.Messaging.Sagas.InMemorySagaPersistor _inMemorySagaPersistor;
        private readonly Foo.Core.IQueryObject<TypeA> _queryObject;
        private readonly Foo.Core.IQueryObject<TypeB> _queryObject;
        
        public Foo_Commands_Command1(Jasper.Messaging.Sagas.InMemorySagaPersistor inMemorySagaPersistor, Foo.Core.IQueryObject<TypeA> queryObject, Foo.Core.IQueryObject<TypeB> queryObject)
        {
            _inMemorySagaPersistor = inMemorySagaPersistor;
            _queryObject = queryObject;
            _queryObject = queryObject;
        }

        public override async Task Handle(Jasper.Messaging.IMessageContext context)
        {
            var saga = new Foo.SampleSaga();
            var command1 = (Foo.Commands.Command1)context.Envelope.Message;
            var inMemoryEnvelopeTransaction = new Jasper.Messaging.Durability.InMemoryEnvelopeTransaction();
            await context.EnlistInTransaction(inMemoryEnvelopeTransaction);
            var sagaStateDocument = saga.Starts(Command1, _queryObject, _queryObject);
            context.EnlistInSaga(sagaStateDocument.Id);
            if (saga.IsCompleted)
            {
                _inMemorySagaPersistor.Delete<Foo.SagaStateDocument>(sagaStateDocument.Id);
            }

            else
            {
                _inMemorySagaPersistor.Store(sagaStateDocument);
            }

        }
```

**Comment by fab60** [comment]
This probably seems to be an issue on the Lamar side: I got the compilation failure if I dynamically register the dependency of the handler method as

```
await _sagaTestHarness.Configure(_ =>
            {
                var queryTypeA = GetQueryObject<TypeA>();
                var queryTypeB = GetQueryObject<TypeB>();
                var queryTypeC = GetQueryObject<TypeC>();

                _.AddSingleton<IQueryObject<TypeA>>(queryTypeA);
                _.AddSingleton<IQueryObject<TypeB>>(queryTypeB);
                _.AddSingleton<IQueryObject<TypeB>>(queryTypeB);
            });
```

as a workaround this instead is successful:

```
await _sagaTestHarness.Configure(_ =>
            {
                var queryTypeA = GetQueryObject<TypeA>();
                var queryTypeB = GetQueryObject<TypeB>();
                var queryTypeC = GetQueryObject<TypeC>();

                _.AddSingleton<IQueryObject<TypeA>>(queryTypeA);
                _.AddSingleton<IQueryObject<TypeB>>(queryTypeB);
                _.AddTransient(__ => queryTypeC);
            });
```

**Comment by jeremydmiller** [comment]
Oh wow, somebody actually using this! What version of Lamar were you on? I've dealt w/ a couple issues like that recently.

**Comment by fab60** [comment]
Hi Jeremy, it should be the latest (2.0)

Overall the experience has been great so far and, as we are evaluating some of the more obvious choices in that space, Jasper has really catched my attention lately

**Comment by jeremydmiller** [comment]
Can I get you to try Lamar 2.0.1? And cool, I'm back to working on it a little bit just today and that's how I finally noticed this issue

**Comment by fab60** [comment]
Actually Jasper 0.9.1 has a dependency on Lamar 1.1.2 and if I try to manually add a Lamar 2.0.1 package, I get an assembly load failure.  

```
"Jasper/0.9.1": {
        "dependencies": {
          "Baseline": "1.5.0",
          "Ben.Demystifier": "0.1.1",
          "Lamar": "1.1.2",
          "Microsoft.AspNetCore.Hosting": "2.1.0",
          "Microsoft.AspNetCore.Hosting.Abstractions": "2.1.0",
          "Microsoft.AspNetCore.Http": "2.1.0",
          "Microsoft.AspNetCore.Http.Abstractions": "2.1.0",
          "Microsoft.AspNetCore.WebUtilities": "2.1.0",
          "Microsoft.Extensions.Caching.Memory": "2.1.0",
          "Microsoft.Extensions.Configuration": "2.1.0",
          "Microsoft.Extensions.Configuration.Abstractions": "2.1.0",
          "Microsoft.Extensions.Configuration.Binder": "2.1.0",
          "Microsoft.Extensions.Configuration.Json": "2.1.0",
          "Microsoft.Extensions.Logging.Console": "2.1.0",
          "Microsoft.Extensions.Logging.Debug": "2.1.0",
          "Newtonsoft.Json": "11.0.2",
          "Oakton": "1.5.0",
          "System.ComponentModel": "4.3.0",
          "System.Net.NameResolution": "4.3.0",
          "System.Threading.Tasks.Dataflow": "4.9.0"
        },
        "runtime": {
          "lib/netstandard2.0/Jasper.dll": {
            "assemblyVersion": "0.0.0.0",
            "fileVersion": "0.0.0.0"
          }
        }
      },
```

**Comment by fab60** [comment]
That’s perfect. Thanks again for your work.

**Comment by jeremydmiller** [comment]
Still working on this, just took longer than I'd hoped

**Comment by jeremydmiller** [comment]
@fab60 I'm going to claim this is fixed by Jasper v0.9.8 (really Lamar 2.0).
